### PR TITLE
Add VCF export support for SNP call datasets

### LINKED ISSUE #1054: Adding other file formats option
All the data that the API produces and uses during its analyses is in the Zarr format (https://zarr.readthedocs.io/en/stable/). However, many common external tools (e.g., ADMIXTURE, bcftools, ...) use other file format. Additionally, many genomic research groups use R (https://www.r-project.org) for many of their analyses and Zarr doesn't have modules for use in R. 

In the interest of expanding the options for our partners, it would be helpful if some of our data could be exported into other commonly used formats (e.g., PLINK, VCF, ...). Some limited support for Plink-conversions already exists but it could certainly be extended to other types of data.

### PR DESCRIPTION
**Pull request by adilraza99**
## Summary
Adds support for exporting SNP call datasets to Variant Call Format (VCF), enabling interoperability with common genomics tools and workflows.

## Changes
- Introduce a `VcfExporter` mixin following the existing `PlinkConverter` pattern
- Implement `snp_calls_to_vcf()` to export SNP calls directly to VCF
- Stream data in chunks to avoid loading the full dataset into memory
- Integrate the exporter into `AnophelesDataResource`
- Add tests following the `test_plink_converter` pattern to verify output structure and dataset consistency

## Notes
The implementation uses `snp_calls()` as the data source so that multiallelic sites are preserved. VCF records are written incrementally from chunked genotype data to keep memory usage low when working with large datasets.

---
closes #1054

### CODE REVIEW

File: malariagen_data/anoph/to_vcf.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+        with opener(output_path, "wt") as f:
+            # Write VCF header.
+            f.write("##fileformat=VCFv4.3\n")
+            f.write(f"##fileDate={date.today().strftime('%Y%m%d')}\n")
+            f.write("##source=malariagen_data\n")
+            for contig in contigs:
+                f.write(f"##contig=<ID={contig}>\n")
+            f.write('##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">\n')
+            header_cols = [
```
Comment (jonbrenas) [inline malariagen_data/anoph/to_vcf.py]: Why were these columns specifically chosen?

File: tests/anoph/test_snp_frq.py [hunk 2]
```diff
@@ -986,18 +986,25 @@ def check_snp_allele_frequencies_advanced(
...
+            period_by=period_by,
+            sample_sets=sample_sets,
+            sample_query=sample_query,
+            sample_query_options=sample_query_options,
+            min_cohort_size=min_cohort_size,
+            nobs_mode=nobs_mode,
+            variant_query=variant_query,
+            site_mask=site_mask,
+        )
+    except ValueError as e:
```
[the remark this answers is not in the document]
Comment (adilraza99) [reply tests/anoph/test_snp_frq.py]: This was added to handle a flaky edge case in the tests.

The test functions use randomized parameters (e.g. sample_sets, cohorts, min_cohort_size), and some combinations can result in zero valid cohorts. In those cases the implementation correctly raises a ValueError ("No cohorts available").

This guard prevents those cases from causing unexpected test failures, while still allowing unexpected errors to surface. I can add a short comment in the test to make this clearer.
